### ISSUE TITLE
Docker optimization

### ISSUE THREAD

**Issue by junghanw** [srvk/DiViMe#40]
1. ```bc: command not found```

```
/home/vagrant/tools/check_folder.sh: line 28: bc: command not found
/vagrant/data/CK67_020227a.wav --> 0.693451
/vagrant/data/CK67_020227a.wav seems empty !```

divime git:(master) ✗ vagrant ssh
Last login: Thu Oct  4 18:03:33 2018 from 172.17.0.1
vagrant@abd158270c0d:~$ bc
-bash: bc: command not found
vagrant@abd158270c0d:~$
```

**Comment by riebling** [comment]
This might be an easy fix: add `bc` to the Dockerfile and test after generating a new image/container, so much so that I assigned it, then you can score the first closed Issue for the directed study team

**Comment by junghanw** [comment]
I realized that vagrant build time is taking too long because of reinstalling all those libraries. So I'm thinking on creating docker image that contains all necessary libraries inside which let users just download the docker image (preferably from docker hub) and get system running using either docker daemon or vagrant provisioning. (users don't have to install docker on their host machine since vagrant takes care of that layer)

I think above approach has several advantages, such as 
 - end-users don't have to care about anything else other than downloading docker image
 - prevents "system fragmentation" among different runtimes and libraries as end-users are not building their own system. 
 - versioning of system can be done using docker image tags 
 - might have better resource management compared to virtualbox. (depends on how vagrant works) 

Things to consider 
 - HTK (using vagrant, we can push HTK package into container in provisioning process, but not so sure how to do this when end-user decides to use docker daemon)
 - size of final docker image (thinking of using multi-stage build, alpine base images, and some optimizations)

**Comment by riebling** [comment]
Interesting to re-visit decisions we made when first building virtual machines, and opted to go with Vagrant so that there can be a very small initial download. (that was the decision at the time) I wonder if we can think more about this in terms of goals; minimizing overall clock time, or as you mentioned 'fragmentation' as something I used to casually think of as "the number of things that can go wrong" when following long provisioning scripts that make numerous downloads and network connections. More on this later, point by point, just wanted to drop my thoughts quickly here.
